packages/flutter/lib/src/widgets/selection.dart

```diff
@@ -0,0 +1,586 @@
+          if (result == SelectionResult.previous) {
+            currentSelectionIndex -= 1;
+            return SelectionResult.end;
+          }
+          break;
+        case SelectionResult.previous:
+          if (currentSelectionIndex == 0) {
+            return result;
+          }
+          currentSelectionIndex -= 1;
+          result = dispatchSelectionEventToChild(selectables[currentSelectionIndex], event);
+          if (result == SelectionResult.end ||
+              result == SelectionResult.pending) {
+            return result;
+          }
+          if (result == SelectionResult.next) {
+            currentSelectionIndex += 1;
+            return SelectionResult.end;
+          }
+          break;
+        case SelectionResult.none:
+          // DragSelectionUpdate should never return none.
+          assert(false);
+          return SelectionResult.none;
+      }
+    }
+    return result;
+  }
+}
+
+/// A selection updater that assumes the selectable child changes infrequently
+/// and without a specific rule.
+///
+/// The updater reset the selection every time a new selectable has been added
+/// or removed from the selectables.
+class StaticMultiSelectableSelectionUpdater extends MultiSelectableSelectionUpdaterBase {
+  @override
+  List<Selectable> get selectables => _selectables;
+  List<Selectable> _selectables = const <Selectable>[];
+
+  DragSelectionStartEvent? _currentDragStart;
+
+  /// Updates the selectables this updater manages.
+  ///
+  /// This method will reset the ongoing selection.
+  void updateSelectables(List<Selectable> other) {
+    if (_selectables == other) {
+      return;
+    }
+    if (!listEquals<Selectable>(_selectables, other)) {
+      // Make sure the existing selectable finishes the drag selection
+      if (_currentDragStart != null) {
+        for (final Selectable selectable in _selectables) {
+          dispatchSelectionEventToChild(selectable, const DragSelectionEndEvent());
+        }
+      }
+      clear();
+      if (_currentDragStart != null) {
+        for (final Selectable selectable in _selectables) {
+          dispatchSelectionEventToChild(selectable, _currentDragStart!);
+        }
+      }
+    }
+    _selectables = other;
+  }
+
+  @override
+  SelectionResult dispatchSelectionEvent(SelectionEvent event) {
+    if (event is DragSelectionStartEvent) {
+      _currentDragStart = event;
+    } else if (event is DragSelectionEndEvent) {
+      _currentDragStart = null;
+    }
+    return super.dispatchSelectionEvent(event);
+  }
+
+}
+
+/// A container that collects the [Selectable]s in the subtree.
+///
+/// This widget creates its own [SelectionContainer] to collect the
+/// selectables in the subtree. These selectables are not visible to any other
+/// [SelectionContainer] above this widget. The creator of this widget
+/// can extract the selectables this container collected using the
+/// [onSelectablesChange] callback and is responsible for the selection update
```

Comment:
will added some doc example, the sole purpose of using SelectionContainer with enabled=true is that you want to manage the selectables subtree by yourself, it doesn't make sense to use the  SelectionContainer without providing the callback.

I have thought about different alternative

1. use globalkey to grab selectables from SelectionContainerState, but it may be hard to detect when the selectables has changed.
2. Use the Inherited way to get the selectable from SelectionContainer, but that means the widget that actually updates the selectables needs to be below the SelectionContainer, it makes code very confusing.